messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalPooledConnectionFactoryAdd.java

```diff
@@ -0,0 +1,206 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.messaging.activemq.jms;
+
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.JGROUPS_CLUSTER;
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.LOCAL;
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.LOCAL_TX;
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.NONE;
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.NO_TX;
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.XA_TX;
+import static org.wildfly.extension.messaging.activemq.MessagingServices.isSubsystemResource;
+import static org.wildfly.extension.messaging.activemq.jms.ConnectionFactoryAttribute.getDefinitions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.activemq.artemis.api.core.DiscoveryGroupConfiguration;
+import org.apache.activemq.artemis.api.core.TransportConfiguration;
+
+import org.jboss.as.connector.metadata.deployment.ResourceAdapterDeployment;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.naming.deployment.ContextNames;
+import org.jboss.as.naming.deployment.ContextNames.BindInfo;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.wildfly.extension.messaging.activemq.CommonAttributes;
+import org.wildfly.extension.messaging.activemq.DiscoveryGroupDefinition;
+import org.wildfly.extension.messaging.activemq.MessagingServices;
+import org.wildfly.extension.messaging.activemq.TransportConfigOperationHandlers;
+import org.wildfly.extension.messaging.activemq.jms.ConnectionFactoryAttributes.Common;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2018 Red Hat, inc.
+ */
+public class ExternalPooledConnectionFactoryAdd extends AbstractAddStepHandler {
+
+    public static final ExternalPooledConnectionFactoryAdd INSTANCE = new ExternalPooledConnectionFactoryAdd();
+
+    private ExternalPooledConnectionFactoryAdd() {
+        super(getDefinitions(ExternalPooledConnectionFactoryDefinition.ATTRIBUTES));
+    }
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
```

Comment:
isn't it possible to share most of this code with PooledConnectionFactoryAdd and only specify what is actually different?